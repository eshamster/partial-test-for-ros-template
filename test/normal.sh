#!/bin/bash

set -eux

root_dir="$(readlink -f $(dirname ${0}))/.."
. "${root_dir}/setenv"

test_name=test-normal
init_dir="${WORK_DIR}/init_normal"

rm -rf "${init_dir}"
mkdir -p "${init_dir}"

function ros_t()
{
    ros template $@
}

ros_t deinit ${test_name} > /dev/null 2>&1 && :

echo "--- create template ---"
ros_t init ${test_name}
ros_t checkout ${test_name}

(
    cd "${WORK_DIR}"
    echo "sample_{{name}}" > file_normal
    echo "sample_{{name}}_{{self}}" > file_djula
    echo "sample" > file_rename

    ros_t add file_normal

    ros_t add file_djula
    ros_t type djula file_djula

    ros_t add file_rename
    ros_t rewrite file_rename "rename_{{name}}.txt"

    echo "- check content -"
    ros_t list
    ros_t cat file_normal
)

echo "--- init using the template ---"
(
    cd "${init_dir}"
    ls
    ros init ${test_name} sample --self this
    echo "- check content -"
    ls | sort
    ls | sort | xargs head
)

echo "--- remove item ---"
(
    ros_t rm file_djula
    ros_t list
)

echo "--- cleanup ---"
ros_t deinit ${test_name} 2>&1 && :
