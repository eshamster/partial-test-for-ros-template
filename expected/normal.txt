--- create template ---
- check content -
      copy  file_rename -> "rename_{{name}}.txt"
      djula file_djula 
      copy  file_normal 
sample_{{name}}
--- init using the template ---
- check content -
file_djula
file_normal
rename_sample.txt
==> file_djula <==
sample_sample_this

==> file_normal <==
sample_{{name}}

==> rename_sample.txt <==
sample
--- remove item ---
      copy  file_rename -> "rename_{{name}}.txt"
      copy  file_normal 
--- cleanup ---
+ ros template deinit test-normal
